api/src/identity-access-management/domain/services/oidc-authentication-service-registry.js

```diff
@@ -36,19 +32,16 @@ export class OidcAuthenticationServiceRegistry {
     return this.#allOidcProviderServices;
   }
 
-  getReadyOidcProviderServices() {
-    return this.#readyOidcProviderServices;
-  }
-
-  getReadyOidcProviderServicesForPixAdmin() {
-    return this.#readyOidcProviderServicesForPixAdmin;
+  getReadyOidcProviderServicesByRequestedApplication(requestedApplication) {
+    return this.#readyOidcProviderServicesByRequestedApplications[
+      generateGroupByKey(requestedApplication.applicationName, requestedApplication.applicationTld)
+    ];
   }
 
   getOidcProviderServiceByCode({ identityProviderCode, requestedApplication }) {
-    const services = requestedApplication?.isPixAdmin
-      ? this.#readyOidcProviderServicesForPixAdmin
-      : this.#readyOidcProviderServices;
-    const oidcProviderService = services.find((service) => identityProviderCode === service.code);
+    const oidcProviderService = this.#readyOidcProviderServicesByRequestedApplications[
+      generateGroupByKey(requestedApplication.applicationName, requestedApplication.applicationTld)
+    ].find((service) => identityProviderCode === service.code);
```

Comment:
On peut réutiliser `this.getReadyOidcProviderServicesByRequestedApplication`

```suggestion
    const appOidcProviderServices = this.getReadyOidcProviderServicesByRequestedApplication(requestedApplication);
    const oidcProviderService = appOidcProviderServices.find((service) => identityProviderCode === service.code);
```

Reply:
FAIT : Code réécrit comme suggéré.